### ISSUE TITLE
Version control in preview

### ISSUE THREAD

**Issue by jesmrec** [owncloud/ios-app#178]
In the file preview, several options to do over the file are available in the three-dot-button. 

By selected "open In", file is downloaded again, so we are assuring the latest version is fetched. Other options could be considered to avoid download again, saving bandwidth or even time if the file is very big.

Checking etag, file dates... 

Share your thoughts

**Comment by felix-schwarz** [comment]
The issue was caused by `UITableView` not doing anything when `reloadData()` is called while the table view isn't visible, and `ClientQueryViewController` actions depending on `OCItem` instances stored in table view cells, which are outdated because of the former, and therefore missing the information that a local copy already exists.

Fixed as of 
https://github.com/owncloud/ios-app/commit/d44bde2793d25e72086bb5f5f3622281606d0fa0.

**Comment by jesmrec** [comment]
The issue is also reproducible in the commit. Maybe i did not explain myself correctly, i will detail deeply:

1. Download a file by tapping (the larger, the better):

![dec-12-2018 09-00-30](https://user-images.githubusercontent.com/14894746/49855112-80370a00-fdec-11e8-8381-021047402b3b.gif)

2. File is previewed correctly
3. Tap in open in... :

![dec-12-2018 09-00-37](https://user-images.githubusercontent.com/14894746/49855119-86c58180-fdec-11e8-95ed-1222b8dd8bc8.gif)

A 37MB file that was downloaded some seconds before in step 1., was downloaded again in step 3., wasting time & bandwidth

Options:

- Keep such behaviour. In the "Open In", always will be included the latest version with the lates changes.
- If the file is locally downloaded and is not downloaded again, we are in the risk of opening the file losing the last changes, but saving time or bandwidth. (check mi comment below)

Solution in the middle: check the etag, file dates or whatever, to download the file only when new changes are available.

Btw, i detected this weird behaviour related with my previous comments:

1. Open a file in the app
2. In web UI, modify the file 
3. In app, "Open In"

Current: Moved to file list, if you tap on any file, it is not downloaded anymore
Expected: to decide with the mentioned options, but after step 3, it should keep on preview and not moving to files view

**Comment by felix-schwarz** [comment]
Thanks for the additional explanation. I indeed missed that part.

The problem there right now is this:
- the viewer starts with the latest `OCItem`
- when the `DisplayViewController` triggers a download, it doesn't update its item with the latest version (now including the info that a local copy exists).
- when the user now selects an action, the outdated copy of the item will be passed to the action

I've fixed this particular bug in https://github.com/owncloud/ios-app/pull/190/commits/735ab11eda08f5c117077ebf8d0e7f349d91f0b0, but a definitive fix will require a bit of a rework of `DisplayViewController` to use a single-item `OCQuery` to keep its item up-to-date.

We need to do this anyway to solve other issues like:
- updating the title in the navigation bar to match the latest name after a rename
- responding to updates of the file contents while it is viewed (via f.ex. the File Provider)

I filed #191 to document and track that.
